[Bug] FE failed to load image when doing checkpoint or restart.

**Describe the bug**
Error in fe.out

```
java.io.IOException: Read transaction states failed
        at org.apache.doris.transaction.GlobalTransactionMgr.readFields(GlobalTransactionMgr.java:371)
        at org.apache.doris.catalog.Catalog.loadTransactionState(Catalog.java:1798)
        at org.apache.doris.catalog.Catalog.loadImage(Catalog.java:1436)
        at org.apache.doris.master.Checkpoint.runAfterCatalogReady(Checkpoint.java:100)
        at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58)
        at org.apache.doris.common.util.Daemon.run(Daemon.java:116)
Caused by: org.apache.doris.common.AnalysisException: errCode = 2, detailMessage = databaseTransactionMgr[14981] does not exist
        at org.apache.doris.transaction.GlobalTransactionMgr.getDatabaseTransactionMgr(GlobalTransactionMgr.java:76)
        at org.apache.doris.transaction.GlobalTransactionMgr.readFields(GlobalTransactionMgr.java:367)
        ... 5 more
```

**To Reproduce**
1. create database and table
2. do some load
3. drop the db
4. make a checkpoint meta image
5. start fe from this image, failed

Introduced by #3369

--- a/fe/src/main/java/org/apache/doris/catalog/Catalog.java
+++ b/fe/src/main/java/org/apache/doris/catalog/Catalog.java
@@ -1803,10 +1803,15 @@ public class Catalog {
 
     public long loadRecycleBin(DataInputStream dis, long checksum) throws IOException {
         if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_10) {
-            Catalog.getCurrentRecycleBin().readFields(dis);
+            recycleBin.readFields(dis);
             if (!isCheckpointThread()) {
                 // add tablet in Recycle bin to TabletInvertedIndex
-                Catalog.getCurrentRecycleBin().addTabletToInvertedIndex();
+                recycleBin.addTabletToInvertedIndex();
+            }
+            // create DatabaseTransactionMgr for db in recycle bin.
+            // these dbs do not exist in `idToDb` of the catalog.
+            for (Long dbId : recycleBin.getAllDbIds()) {
+                globalTransactionMgr.addDatabaseTransactionMgr(dbId);
             }
         }
         return checksum;

--- a/fe/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java
+++ b/fe/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java
@@ -174,7 +174,7 @@ public class CatalogRecycleBin extends MasterDaemon implements Writable {
                 idToRecycleTime.remove(entry.getKey());
 
                 // remove database transaction manager
-                Catalog.getInstance().getGlobalTransactionMgr().removeDatabaseTransactionMgr(db.getId());
+                Catalog.getCurrentCatalog().getGlobalTransactionMgr().removeDatabaseTransactionMgr(db.getId());
 
                 LOG.info("erase database[{}] name: {}", db.getId(), dbName);
             }

--- a/fe/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java
+++ b/fe/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java
@@ -173,6 +173,9 @@ public class CatalogRecycleBin extends MasterDaemon implements Writable {
                 iterator.remove();
                 idToRecycleTime.remove(entry.getKey());
 
+                // remove database transaction manager
+                Catalog.getInstance().getGlobalTransactionMgr().removeDatabaseTransactionMgr(db.getId());
+
                 LOG.info("erase database[{}] name: {}", db.getId(), dbName);
             }
         }

--- a/fe/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java
+++ b/fe/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java
@@ -36,6 +36,7 @@ import org.apache.doris.task.DropReplicaTask;
 import org.apache.doris.thrift.TStorageMedium;
 
 import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Range;
 import com.google.common.collect.Sets;
@@ -886,5 +887,10 @@ public class CatalogRecycleBin extends MasterDaemon implements Writable {
                 isInMemory = in.readBoolean();
             }
         }
+    }
+    
+    // currently only used when loading image. So no synchronized protected.
+    public List<Long> getAllDbIds() {
+        return Lists.newArrayList(idToDatabase.keySet());
     }
 }

--- a/fe/src/main/java/org/apache/doris/qe/AuditEventProcessor.java
+++ b/fe/src/main/java/org/apache/doris/qe/AuditEventProcessor.java
@@ -56,6 +56,7 @@ public class AuditEventProcessor {
 
     public void start() {
         workerThread = new Thread(new Worker(), "AuditEventProcessor");
+        workerThread.setDaemon(true);
         workerThread.start();
     }
 

--- a/fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java
+++ b/fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java
@@ -79,11 +79,15 @@ public class GlobalTransactionMgr implements Writable {
     }
 
     public void addDatabaseTransactionMgr(Long dbId) {
-        dbIdToDatabaseTransactionMgrs.putIfAbsent(dbId, new DatabaseTransactionMgr(dbId, catalog, idGenerator));
+        if (dbIdToDatabaseTransactionMgrs.putIfAbsent(dbId, new DatabaseTransactionMgr(dbId, catalog, idGenerator)) == null) {
+            LOG.debug("add database transaction manager for db {}", dbId);
+        }
     }
 
     public void removeDatabaseTransactionMgr(Long dbId) {
-        dbIdToDatabaseTransactionMgrs.remove(dbId);
+        if (dbIdToDatabaseTransactionMgrs.remove(dbId) != null) {
+            LOG.debug("remove database transaction manager for db {}", dbId);
+        }
     }
 
     public long beginTransaction(long dbId, List<Long> tableIdList, String label, TxnCoordinator coordinator, LoadJobSourceType sourceType,
@@ -355,19 +359,19 @@ public class GlobalTransactionMgr implements Writable {
     }
     
     public void readFields(DataInput in) throws IOException {
-        try {
-            int numTransactions = in.readInt();
-            for (int i = 0; i < numTransactions; ++i) {
-                TransactionState transactionState = new TransactionState();
-                transactionState.readFields(in);
+        int numTransactions = in.readInt();
+        for (int i = 0; i < numTransactions; ++i) {
+            TransactionState transactionState = new TransactionState();
+            transactionState.readFields(in);
+            try {
                 DatabaseTransactionMgr dbTransactionMgr = getDatabaseTransactionMgr(transactionState.getDbId());
                 dbTransactionMgr.unprotectUpsertTransactionState(transactionState, true);
+            } catch (AnalysisException e) {
+                LOG.warn("failed to get db transaction manager for txn: {}", transactionState);
+                throw new IOException("Read transaction states failed", e);
             }
-            idGenerator.readFields(in);
-        } catch (AnalysisException e) {
-            throw new IOException("Read transaction states failed", e);
         }
-
+        idGenerator.readFields(in);
     }
 
     public TransactionState getTransactionStateByCallbackIdAndStatus(long dbId, long callbackId, Set<TransactionStatus> status) {

